### ISSUE TITLE
Can't name a model data node as called `data`

### ISSUE THREAD

**Issue by danielturek** [nimble-dev/nimble#941]

**Comment by danielturek** [comment]
If you have a node in a model named `"data"`, then you can't assign it to be a data node, in particular if its **not** the last element provided in the `data` list.

Anyway, the error results from the statement:
```
assign(varName, varValue, inherits = TRUE)
```
which over-writes the `data` list itself, in `BUGS_model.R`.

Reproducible example:

```
library(nimble)

code <- nimbleCode({
    data ~ dnorm(0, 1)
    y ~ dnorm(0, 1)
})

dataList <- list(data = 1, y = 1)

Rmodel <- nimbleModel(code, data = dataList)

## Error in data[[varName]] : subscript out of bounds
```

**Comment by paciorek** [comment]
Fixed by PR #946
